test/integration/deployment/util.go

```diff
@@ -191,13 +196,67 @@ func (d *deploymentTester) waitForDeploymentStatusValid() error {
 
 // waitForDeploymentStatusValidAndMarkPodsReady waits for the Deployment status to become valid
 // while marking all Deployment pods as ready at the same time.
-func (d *deploymentTester) waitForDeploymentStatusValidAndMarkPodsReady() {
+func (d *deploymentTester) waitForDeploymentStatusValidAndMarkPodsReady() error {
 	// Manually mark all Deployment pods as ready in a separate goroutine
 	go d.markAllPodsReady()
 
 	// Make sure the Deployment status is valid while Deployment pods are becoming ready
 	err := d.waitForDeploymentStatusValid()
 	if err != nil {
-		d.t.Fatalf("failed to wait for Deployment status %s: %v", d.deployment.Name, err)
+		return fmt.Errorf("failed to wait for Deployment status %s: %v", d.deployment.Name, err)
+	}
+	return nil
+}
+
+func (d *deploymentTester) updateDeployment(applyUpdate testutil.UpdateDeploymentFunc) (*v1beta1.Deployment, error) {
+	return testutil.UpdateDeploymentWithRetries(d.c, d.deployment.Namespace, d.deployment.Name, applyUpdate, d.t.Logf)
+}
+
+func (d *deploymentTester) waitForObservedDeployment(desiredGeneration int64) error {
+	if err := testutil.WaitForObservedDeployment(d.c, d.deployment.Namespace, d.deployment.Name, desiredGeneration); err != nil {
+		return fmt.Errorf("failed waiting for ObservedGeneration of deployment %s to become %d: %v", d.deployment.Name, desiredGeneration, err)
+	}
+	return nil
+}
+
+func (d *deploymentTester) getNewReplicaSet() (*v1beta1.ReplicaSet, error) {
+	rs, err := deploymentutil.GetNewReplicaSet(d.deployment, d.c.ExtensionsV1beta1())
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving new replicaset of deployment %s: %v", d.deployment.Name, err)
+	}
+	return rs, nil
+}
+
+func (d *deploymentTester) expectNoNewReplicaSet() error {
+	rs, err := d.getNewReplicaSet()
+	if err != nil {
+		return err
+	}
+	if rs != nil {
+		return fmt.Errorf("expected deployment %s not to create a new replicaset, got %v", d.deployment.Name, rs)
+	}
+	return nil
+}
+
+func (d *deploymentTester) expectNewReplicaSet() (*v1beta1.ReplicaSet, error) {
+	rs, err := d.getNewReplicaSet()
+	if err != nil {
+		return nil, err
+	}
+	if rs == nil {
+		return nil, fmt.Errorf("expected deployment %s to create a new replicaset, got nil", d.deployment.Name)
+	}
+	return rs, nil
+}
+
+func pauseFn() func(update *v1beta1.Deployment) {
+	return func(update *v1beta1.Deployment) {
```

Comment:
Could this just be `var pauseFn = func(...`?

Reply:
Nevermind. Not worth changing since tests already passed. :)

Reply:
I can update this in the next integration test PR